src/emit/binary.rs

```diff
@@ -1015,6 +1015,7 @@ impl<'a> Binary<'a> {
                 Type::FunctionSelector => {
                     self.llvm_type(&Type::Bytes(ns.target.selector_length()), ns)
                 }
+                Type::Void => BasicTypeEnum::IntType(self.context.i64_type()),
```

Comment:
There is a llvm/inkwell `self.context.void_type().into()` for this.

Reply:
Soroban functions always return a value encoded in 64 bits. If functions return type is void, that means it returns `2_u64`. However, I should have explained this in a comment

Reply:
That is Soroban specific, and that doesn't belong in the llvm type system.

What's the return value? Is it a success/failure code? That's the same as Solana and Polkadot, and that's implemented in the function emit code.

Reply:
@seanyoung The return value is not a success/failure, it is just the returned data
If no data is returned, it returns void (which is a uint64 in Soroban)